x-pack/plugins/maps/public/elasticsearch_geo_utils.js

```diff
@@ -205,14 +206,63 @@ export function createExtentFilter(mapExtent, geoFieldName, geoFieldType) {
       }
     };
   } else {
-    const errorMessage = i18n.translate('xpack.maps.elasticsearch_geo_utils.unsupportedGeoFieldTypeErrorMessage', {
+    const errorMessage = i18n.translate('xpack.maps.elasticsearch_geo_utils.extent.unsupportedGeoFieldTypeErrorMessage', {
       defaultMessage: `Unsupported field type, expected: geo_shape or geo_point, you provided: {geoFieldType}`,
       values: { geoFieldType }
     });
     throw new Error(errorMessage);
   }
 }
 
+export function createShapeFilter(geojsonPolygon, indexPatternId, geoFieldName, geoFieldType) {
+
+  //take outer
+  const points  = geojsonPolygon.coordinates[0].map(coordinatePair => {
+    return {
+      lon: _.round(coordinatePair[0], DECIMAL_DEGREES_PRECISION),
+      lat: _.round(coordinatePair[1], DECIMAL_DEGREES_PRECISION)
+    };
+  });
+
+  const filter = {
+    meta: {
+      negate: false,
+      index: indexPatternId,
+      alias: i18n.translate('xpack.maps.elasticsearch_geo_utils.filter.aliasTitle', {
+        defaultMessage: `geo polygon at {coordinate}`,
+        // eslint-disable-next-line max-len
+        values: { coordinate: `${_.round(geojsonPolygon.coordinates[0][0][0], DECIMAL_DEGREES_PRECISION)}, ${_.round(geojsonPolygon.coordinates[0][0][1], DECIMAL_DEGREES_PRECISION)}` }
+      })
+    }
+  };
+
+  if (geoFieldType === ES_GEO_FIELD_TYPE.GEO_POINT) {
+    filter.geo_polygon = {
+      ignore_unmapped: true,
+      [geoFieldName]: {
+        points: points
+      }
+    };
+  } else if (geoFieldType === ES_GEO_FIELD_TYPE.GEO_SHAPE) {
+    filter.geo_shape = {
+      ignore_unmapped: true,
+      [geoFieldName]: {
+        shape: geojsonPolygon,
```

Comment:
Use `points` to be consistent. geojsonPolygon is updated in place but might as well use `points` like the bounding box filter